groups/bsl/bslstl/bslstl_ratio.t.cpp

```diff
@@ -0,0 +1,383 @@
+static bool veryVeryVerbose;
+static bool veryVeryVeryVerbose;
+
+//=============================================================================
+//                  GLOBAL HELPER FUNCTIONS FOR TESTING
+//-----------------------------------------------------------------------------
+
+//=============================================================================
+//                              MAIN PROGRAM
+//-----------------------------------------------------------------------------
+
+int main(int argc, char *argv[])
+{
+    int            test = argc > 1 ? atoi(argv[1]) : 0;
+                verbose = argc > 2;
+            veryVerbose = argc > 3;
+        veryVeryVerbose = argc > 4;
+    veryVeryVeryVerbose = argc > 5;
+
+    printf("TEST " __FILE__ " CASE %d\n", test);
+
+    switch (test) { case 0:  // Zero is always the leading case.
+      case 1: {
+        // --------------------------------------------------------------------
+        // TESTING <ratio> HEADER VERIFICATION
+        //
+        // Concerns:
+        //: 1 That the operations are successfully imported into the 'bsl'
+        //:   namespace from the underlying STL implementation.
+        //: 
+        //: 2 That the short versions of the comparison operators, such
+        //:   as 'ratio_less_v' exist and work even if the underlying STL
+        //:   does not provide them.
+        //
+        // Plan:
+        //: 1 Verify that the template 'ratio' exists in the bsl namespace and
+        //:   that it behaves like std::ratio.
+        //:
+        //: 2 Verify that the SI typedefs exist and have the correct numerator
+        //:   and denominator.
+        //:
+        //: 3 Verify that the comparison operations exist, and give expected
+        //:   results. Check again for the '_v' versions.
+        //:
+        //: 4 Verify that the arithmetic operations exist, and give expected
+        //:   results.
+        //
+        // Testing:
+        //  Basic ratio functionality - numerator/denominator/nested 'type'
+        //   SI prefixes - kilo, mega, milli, etc.
+        //   Ratio comparisons - less/greater/equal, etc.
+        //   Ratio operations - add/subtract/multiply/divide, etc.
+        // --------------------------------------------------------------------
+
+        if (verbose) printf("\nTesting <ratio> HEADER VERIFICATION"
+                            "\n===================================\n");
+
+        if (verbose) printf("\tTest basic ratio functionality.\n");
+        {
+            typedef bsl::ratio<0, 1> Zero;
+            typedef bsl::ratio<2> Two;
+            typedef bsl::ratio<6, 4> Six4ths;
+
+            ASSERT(0 == Zero::num);
+            ASSERT(1 == Zero::den);
+            ASSERT(2 == Two::num);
+            ASSERT(1 == Two::den);
+
+            // make sure that the ratio is reduced
+            ASSERT(3 == Six4ths::num);
+            ASSERT(2 == Six4ths::den);
+            
+            // A nested typedef named 'type'
```

Comment:
In BDE, comments begin with [A-Z] iff they end with ".".  (Note that this isn't a complete sentence, so adding "." wouldn't be appropriate here.)
s/typedef/'typedef'/